install.sh

```diff
@@ -45,16 +45,18 @@ chmod +x "$exe"
 rm "$exe.zip"
 
 echo "Deno was installed successfully to $exe"
+if $exe eval 'const [major, minor] = Deno.version.deno.split("."); if (major < 2 && minor < 42) Deno.exit(1)'; then
+
+	allow_sys_perm="--allow-sys=homedir"
+
+	if ! echo | $exe run --allow-sys=homedir - >/dev/null 2>&1; then
+		allow_sys_perm="--allow-sys"
+	fi
+	$exe run $allow_sys_perm --no-check --allow-run="zsh,$exe" --allow-read --allow-env --allow-write="$deno_install" ./shell-setup/src/main.ts "$deno_install"
```

Comment:
Will this part (`./shell-setup/src/main.ts`) be updated to jsr url later?

nit: Is `--no-check` flag necessary?

Reply:
Yup, was planning on updating it to jsr (just did it now). Removed `--no-check` as well.